expression/builtin_cast.go

```diff
@@ -558,3 +817,110 @@ func (b *builtinCastDurationAsTimeSig) evalTime(row []types.Datum) (res types.Ti
 	res, err = res.RoundFrac(b.tp.Decimal)
 	return res, false, errors.Trace(err)
 }
+
+func buildCastFunction(expr Expression, tp *types.FieldType, ctx context.Context) (*ScalarFunction, error) {
+	var fc functionClass
+	switch tp.ToClass() {
+	case types.ClassInt:
```

Comment:
we check it in line854,
buildCastFunction is only responsible for build the `cast` ScalarFunction,
thus can be reused in  NewCastFunction.